fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -740,6 +741,10 @@ private Resource doRead(String type, String id, boolean throwExcOnNull, boolean
 
             FHIRSearchContext searchContext = null;
             if (queryParameters != null) {
+                if (!queryParameters.keySet().stream().allMatch(k -> FHIRConstants.GENERAL_PARAMETER_NAMES.contains(k))) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
```

Comment:
Wondering if we should take into account the 'lenient' vs. 'strict' setting, and only log the issue if in lenient mode.

Reply:
Updated to only log the issue if in lenient mode.